### PR TITLE
[DS-368] restrictMaxLength to allowExceedingMaxLength

### PR DESCRIPTION
**Pull request by victortrinh2**
restrictMaxLength to allowExceedingMaxLength

### CODE REVIEW

File: packages/components/src/inputs/src/TextArea.tsx
Comment (vicky-comeau) [inline packages/components/src/inputs/src/TextArea.tsx:1]: everything looks good, just missing the console.warn. There should already be an example of this somewhere.

File: packages/components/src/inputs/src/TextField.tsx
Comment (vicky-comeau) [inline packages/components/src/inputs/src/TextField.tsx:1]: same comment as above

**Review by alexasselin008 — APPROVED** [review]

File: packages/components/src/inputs/src/TextArea.tsx [hunk 1]
```diff
@@ -64,9 +64,9 @@ export interface TextAreaProps extends StyledComponentProps<RACTextFieldProps> {
      * This should only be used with the `showCharacterCount` prop.
      * If `true`, the TextArea prevents the text from ever going over the max length.
      * If `false`, the TextArea will allow the text to go over the max length, but it will add an error look tot he character count.
-     * @default true
+     * @default false
```
Comment (vicky-comeau) [inline packages/components/src/inputs/src/TextArea.tsx:67]: I think in this case, we no longer need to set a default. The default isn't actually false, it's undefined.

File: packages/components/src/inputs/src/TextField.tsx [hunk 2]
```diff
@@ -66,9 +66,9 @@ export interface TextFieldProps extends StyledComponentProps<RACTextFieldProps>
      * This should only be used with the `showCharacterCount` prop.
      * If `true`, the TextArea prevents the text from ever going over the max length.
      * If `false`, the TextArea will allow the text to go over the max length, but it will add an error look tot he character count.
-     * @default true
+     * @default false
```
Comment (vicky-comeau) [inline packages/components/src/inputs/src/TextField.tsx:69]: same comment as above

File: packages/components/src/inputs/src/TextField.tsx [hunk 3]
```diff
@@ -64,11 +64,9 @@ export interface TextFieldProps extends StyledComponentProps<RACTextFieldProps>
 
     /**
      * This should only be used with the `showCharacterCount` prop.
-     * If `true`, the TextArea prevents the text from ever going over the max length.
-     * If `false`, the TextArea will allow the text to go over the max length, but it will add an error look tot he character count.
-     * @default true
+     * If `true`, the TextArea will allow the text to go over the max length, but it will add an error look to the character count.
```
Comment (vicky-comeau) [inline packages/components/src/inputs/src/TextField.tsx:67]: I guess this was a copy paste error before, sorry. it should says text field here

**Review by vicky-comeau — APPROVED** [review]

**Pull request merged**